src/validation_workflow/rpm/validation_rpm.py

```diff
@@ -25,15 +25,13 @@ def __init__(self, args: ValidationArgs) -> None:
         self.tmp_dir = TemporaryDirectory()
 
     def download_artifacts(self) -> bool:
-        architectures = ["x64", "arm64"]
         for project in self.args.projects:
-            for architecture in architectures:
-                url = f"{self.base_url}{project}/{self.args.version}/{project}-{self.args.version}-linux-{architecture}.rpm"
-                if ValidateRpm.is_url_valid(url) and ValidateRpm.download(url, self.tmp_dir):
-                    logging.info(f"Valid URL - {url} and Download Successful !")
-                else:
-                    logging.info(f"Invalid URL - {url}")
-                    raise Exception(f"Invalid url - {url}")
+            url = f"{self.base_url}{project}/{self.args.version}/{project}-{self.args.version}-linux-{self.args.arch}.rpm"
+            if ValidateRpm.is_url_valid(url) and ValidateRpm.download(url, self.tmp_dir):
+                logging.info(f"Valid URL - {url} and Download Successful !")
+            else:
+                logging.info(f"Invalid URL - {url}")
+                raise Exception(f"Invalid url - {url}")
```

Comment:
Add a `check!` method in the parent class of `ValidateRpm` and `ValidateTar` that does this to avoid duplication.

Reply:
> There are no tests to this code change, which is concerning.
> 
> * Refactor validation checks that are identical.
> * Add tests.

Thanks @dblock for reviewing the PR,
-Just the tarball, rpm and yum distributions use the same implementation for validating url's but not docker, So I had the the methods is_url_valid(url) and download(url, self.tmp_dir) in a seperate parent class and called them individually, I can actually have the complete code block in Function like` Check` to avoid duplication.

- So far the code changes in the PR are to edit the logging statements but does not alter function-calls or the return type of value, So this didn't not require any updates to the tests. I'll update them accordingly along with the code changes if needed.